src/Files.App/Helpers/QuickLookHelpers.cs

```diff
@@ -1,35 +1,68 @@
-using Files.Shared.Extensions;
-using System.Diagnostics;
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Security.Principal;
 using System.Threading.Tasks;
-using Windows.Foundation.Collections;
 
-namespace Files.App.Helpers
+namespace Files.App.Helpers;
+
+public static class QuickLookHelpers
 {
-    public static class QuickLookHelpers
+    public static Task ToggleQuickLook(IShellPage associatedInstance, bool switchPreview = false)
     {
-        public static async Task ToggleQuickLook(IShellPage associatedInstance, bool switchPreview = false)
+        if (!App.AppModel.IsQuickLookSupported || !associatedInstance.SlimContentPage.IsItemSelected || associatedInstance.SlimContentPage.IsRenamingItem)
+            return Task.CompletedTask;
+
+        App.Logger.Info("Toggle QuickLook");
+
+        string PipeName = $"QuickLook.App.Pipe.{WindowsIdentity.GetCurrent().User?.Value}";
+        string Message = switchPreview ? "QuickLook.App.PipeMessages.Switch" : "QuickLook.App.PipeMessages.Toggle";
+
+        using var client = new NamedPipeClientStream(".", PipeName, PipeDirection.Out);
+        client.Connect();
+
+        using var writer = new StreamWriter(client);
+        writer.WriteLine($"{Message}|{associatedInstance.SlimContentPage.SelectedItem.ItemPath}");
+        writer.Flush();
+        return Task.CompletedTask;
+    }
+
+    public static bool CheckQuickLookAvailability()
+    {
+        static int QuickLookServerAvailable()
         {
-            if (!App.AppModel.IsQuickLookSupported || !associatedInstance.SlimContentPage.IsItemSelected || associatedInstance.SlimContentPage.IsRenamingItem)
+            string pipeName = $"QuickLook.App.Pipe.{WindowsIdentity.GetCurrent().User?.Value}";
+            string pipeSwitch = "QuickLook.App.PipeMessages.Switch";
+
+            using var client = new NamedPipeClientStream(".", pipeName, PipeDirection.Out);
+            try
             {
-                return;
-            }
+                client.Connect(500);
```

Comment:
We might want `await client.ConnectAsync(500)` instead. Since you now execute this directly instead of in a separate process Connect() might freeze the UI.

Reply:
I made a bunch of changes to the methods. Added in the await usings as well.

Reply:
Wow `await using` is new to mee, cool.

Reply:
C#8 I believe